CFC_DataCollector/trip_analysis/cluster_pipeline.py

```diff
@@ -0,0 +1,130 @@
+import similarity
+from get_database import get_section_db
+import featurization
+import sys
+import math
+
+"""
+This file reads the data from the section database, 
+removes noise from the data, and clusters is. 
+
+The parameters and clustering methods can be easily changed, 
+but based on what works the best, the featurization and clustering 
+works as follows. First, the data is read from the database. 
+For featurization, each section is representated as a start point 
+and an end point. Then, the section is put into bins and the lower 
+half of the bins are removed. Then, the data is clustered using 
+k-means. The parameter for k is currently tested in a range based 
+on the number of elements, but I plan to adjust this range. 
+
+As input, this file accepts an user's uuid from the command line. 
+
+Currently, this file is defaulted not to handle ground truth, but 
+for the purpose of tests, a parameter can be changed to collect ground 
+truth from the database and compare the clustering to ground truth. 
+To change this, in main, change ground_truth to True in the call to 
+read_data. 
+"""
+
+#read the data from the database. If ground_truth is true, it will 
+#take it from the 'color' field of each section in the database. 
```

Comment:
I didn't add it to the database, but I can. I added it to the database on my computer for testing, and I thought it was an easy way to incorporate it for running tests. I can add it to the server as well for the data that we have ground truthed.